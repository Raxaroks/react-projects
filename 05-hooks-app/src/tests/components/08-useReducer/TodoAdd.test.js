import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import TodoAdd from '../../../components/08-useReducer/TodoAdd';


describe('Pruebas en el componente TodoAdd', () => {

    const handleAddTodo = jest.fn();

    const componentWrapper = shallow(
        <TodoAdd
            handleAddTodo={ handleAddTodo }
        />
    );

    test('Debe de renderizarse de manera correcta.', () => {
        expect(componentWrapper).toMatchSnapshot();
    });

    test('No debe de llamar a la función handleAddTodo.', () => {
        const formSubmit = componentWrapper.find('form').prop('onSubmit');
        formSubmit( { preventDefault(){} } );

        expect(handleAddTodo).toHaveBeenCalledTimes(0);
    });

    test('Debe de llamar a la función handleAddTodo.', () => {
        const value = 'Aprender ReactJS';
        componentWrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const formSubmit = componentWrapper.find('form').prop('onSubmit');
        formSubmit( { preventDefault(){} } );

        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false
        });

        expect( componentWrapper.find('input').prop('value') ).toBe('');
    });

});
