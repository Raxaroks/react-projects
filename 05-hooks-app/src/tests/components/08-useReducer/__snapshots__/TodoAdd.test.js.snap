// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TodoAdd Debe de renderizarse de manera correcta. 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
   
  <hr />
  <form
    className="d-grid gap-2"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
